Documentation/getting-started.md

```diff
@@ -0,0 +1,145 @@
+# Getting Started
+
+If you would like to contribute, but don't know how to start, this page is for
+you. This guide will walk you through running Oreboot locally using QEMU
+emulation and explain the high level architecture / structure of the project.
+It also assumes the use of Debian like system.
+
+## Set up the environment
+
+The first step, is to install all dependencies and ensure the project can be
+started locally (see `README.md` and `azure-pipelines.yml` for possibly more
+up-to-date instructions.
+
+```shell
+git clone git://github.com/oreboot/oreboot
+cd oreboot
+make debiansysprepare
+make firsttime
+```
+
+Next, we need to install QEMU - generic machine emulator that will allow us to
+emulate running oreboot without the need of custom hardware. Below commands
+might be a little outdated, so please look at `azure-pipelines.yml` file for the
+actual commands that are used in CI.
+
+```shell
+# Go to a directory where QEMU will be fetched
+cd ~/
+# Fetch and build QEMU
+git clone https://github.com/qemu/qemu && cd qemu && git checkout <put tag from azure-pipelines.yml>
```

Comment:
I don't think you need to indicate a tag here. It would be best to just use QEMU master.

Reply:
When I was starting with oreboot, HEAD master of QEMU was actually not working. For a new contributor, I wanted to increase the chance of it working for the first time (later they can experiment with other versions of QEMU).

Reply:
Is there anything wrong with just using the distro package? I'd not want to have ti build QEMU myself... and then again, why would CI need to build it? For building from source, I would definitely stick with a tagged and presumably stable release.

Reply:
Distro packages work great, you just need to use a modern distro with v5.1.0 or newer.

Can we update this to tell users to use their distro QEMU if it's v5.1.0 or newer, otherwise build v5.1.0?

Reply:
1. This example shows RISC-V - is there prebuilt qemu for RISC-V that can be downloaded (I though this is the main reason we build it).
2. The exact version is important for RISC-V, because when I started ~month HEAD master was not working, and I needed a specific old version to have it run properly

I created #295 to setup qemu-q35 in CI. Once that is done, I think we could update this guide to just use distro package and use q35 instead of riscv. WDYT?

Reply:
If we add q35 testing then we should test both not just 1. It doesn't make a lot of sense to remove test cases from the CI. RISC-V is stabalising quickly so oreboot should be fine to use 5.1+ for the foreseeable future.

Reply:
Yes, we definitely should still test both and we should not remove RISC-V from CI. What I meant is to remove RISC-V from this guide, and just use q35 (there is no need for the user to mention both if q35 is simpler to start with). And I was suggesting to remove risc-v from this guide because of this comment https://github.com/oreboot/oreboot/pull/289#pullrequestreview-492220337 from Ron.

Reply:
In Arch, we have RISC-V QEMU packaged. 5.1.0 is the current version. See here:
https://www.archlinux.org/packages/extra/x86_64/qemu-arch-extra/